src/domain/enrolment/__tests__/CreateEnrolmentPage.test.tsx

```diff
@@ -397,3 +397,60 @@ test('render and focuses error notification correctly', async () => {
 
   expect(screen.getByRole('textbox', { name: 'Nimi' })).toHaveFocus();
 });
+
+// FIXME: Tested in browser and the result is accepted. Unit test fails.
+test.skip('amount of adults is limited to 1 when there is 1 seat left after children', async () => {
+  render(<CreateEnrolmentPage />, {
+    mocks: mock3,
+    query: { eventId: eventId, occurrences: occurrenceIds },
+  });
+  await waitFor(() => {
+    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  });
+  userEvent.type(screen.getByLabelText(/lapsia/i), '19');
+  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
+  await act(wait);
+  expect(
+    screen.getByText(
+      /Arvon tulee olla enintään 1 yhdessä lasten lukumäärän kanssa/i
+    )
+  ).toBeInTheDocument();
+});
+
+// FIXME: Tested in browser and the result is accepted. Unit test fails.
+test.skip('amount of adults is limited to 0 when children fills the group in whole', async () => {
+  render(<CreateEnrolmentPage />, {
+    mocks: mock3,
+    query: { eventId: eventId, occurrences: occurrenceIds },
+  });
+  await waitFor(() => {
+    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  });
+  userEvent.type(screen.getByLabelText(/lapsia/i), '20');
+  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
+  await act(wait);
+  expect(
+    screen.getByText(
+      /Arvon tulee olla enintään 0 yhdessä lasten lukumäärän kanssa/i
+    )
+  ).toBeInTheDocument();
+});
+
+test('adults should not validated against max group limit when there are already too many children', async () => {
+  render(<CreateEnrolmentPage />, {
+    mocks: mock3,
+    query: { eventId: eventId, occurrences: occurrenceIds },
+  });
+  await waitFor(() => {
+    expect(screen.getByLabelText(/lapsia/i)).toBeInTheDocument();
+  });
+  userEvent.type(screen.getByLabelText(/lapsia/i), '21');
+  userEvent.type(screen.getByLabelText(/aikuisia/i), '2');
+  await act(wait);
```

Comment:
Is there any better alternatives? I tried fireEvents inside an act also, but they didn't work. Example here: https://duncanleung.com/testing-validation-schema-formik-form/

Reply:
Formik does validation is asynchronous manner so this is perfect place to use the waitFor function.

```
  await waitFor(() => {
    expect(
      screen.getByText(
        /Arvon tulee olla enintään 20/i
      )
    ).toBeInTheDocument();
  });
```

Reply:
And of course add `userEvent.tab()` to really test this:

```
 expect(
    screen.queryByText(
      /Arvon tulee olla enintään \d yhdessä lasten lukumäärän kanssa/i
    )
  ).toBeNull();
```

Atm I think it passes always because the input is never unfocused

Reply:
Also `.toBeNull()` -> `.not.toBeInTheDocument()`

Reply:
and when using `.toBeInTheDocument()` it would be better to use screen.queryBy function because it return null if it doesn't find the element. `getBy` functions throws error itself if it doesn't find the element

Reply:
worked! Thank you!